web/src/components/forms/ReactiveInputDate.tsx

```diff
@@ -0,0 +1,50 @@
+﻿import {ObservablePrimitiveChildFns} from "@legendapp/state";
+import {Reactive} from "@legendapp/state/react";
+import {add, format, parse} from "date-fns";
+
+export interface Props {
+    value: ObservablePrimitiveChildFns<string>;
+    label: string;
+    required?: boolean;
+    className?: string;
+}
+
+const ReactiveInputDate = ({value, label, required, className}: Props) => {
+
+    let thinkTimeout: any;
+    value.onChange(() => {
+        if (thinkTimeout) {
+            clearTimeout(thinkTimeout);
+        }
+
+        thinkTimeout = setTimeout(() => formatDate(), 500);
+    });
+
+    function formatDate() {
+        if (value.get().length < 5) {
+            return;
+        }
+
+        const parsed = parse(
+            value.get(),
+            "d/M/yy",
+            add(new Date(), {
+                years: -50,
+            })
+        );
+
+        if (parsed.toString() !== "Invalid Date") {
+            value.set(format(parsed, "dd/MM/yyyy"));
+        }
+    }
+
+    return <div className={"block" + " " + className}>
```

Comment:
[clsx](https://www.npmjs.com/package/clsx)
```suggestion
    return <div className={clsx("block", className)}>
```

Reply:
Connaissait pas, merci :)